ui/src/main/webapp/src/app/reports/migration-issues/migration-issues.service.ts

```diff
@@ -20,10 +23,79 @@ export class MigrationIssuesService extends AbstractService {
             .catch(this.handleError);
     }
 
+    @Cached({section: 'migrationIssues', immutable: true})
+    getDataFromServer(executionId: number) {
+        let url = MigrationIssuesService.NEW_AGGREGATED_ISSUES_URL.replace('{executionId}', executionId.toString());
+
+        return this._http.get(url)
+            .map(res => res.json())
+            .catch(this.handleError);
+    }
+
+    getNewAggregatedIssues(executionId: number, filter?: ReportFilter): Observable<Dictionary<ProblemSummary[]>> {
+        return this.getDataFromServer(executionId)
+            .map((serverResponse: MigrationIssuesResponse[]) => {
+                let result: Dictionary<ProblemSummary[]> = {};
+
+                let data  = serverResponse;
+
+
+                if (filter) {
+                    data = serverResponse.filter(item => {
+                        return filter.selectedApplications.some(filterApp => {
+                            let rootFileName =  (<any>item.projectModel).vertices_out.rootFileModel.vertices[0].fileName;
+
+                            if ((<FilterApplication>filterApp).fileName) {
+                                return rootFileName === (<FilterApplication>filterApp).fileName;
+                            } else {
+                                return rootFileName === (<Partial<RegisteredApplication>>filterApp).inputFilename;
+                            }
+                        });
+                    });
+                }
```

Comment:
@jsight What do you think about this approach to filtering?

We could reuse this part of code also for other reports.
By using cache on `getDataFromServer` method, it only triggers one HTTP request to server to get complete migration issues report. Viewing the full report or report for any application, and switching between them, doesn't cost any request. It uses data from cache and only applies appropriate filters.